tests/serverpod_test_server/test_integration/serverpod_cli_args_override_test.dart

```diff
@@ -0,0 +1,342 @@
+      final config = ServerpodConfig(
+        runMode: 'development',
+        serverId: 'test-server',
+        role: ServerpodRole.monolith,
+        loggingMode: ServerpodLoggingMode.normal,
+        applyMigrations: false,
+        applyRepairMigration: false,
+        apiServer: ServerConfig(
+          port: 8080,
+          publicHost: 'localhost',
+          publicPort: 8080,
+          publicScheme: 'http',
+        ),
+        maxRequestSize: 524288,
+      );
+
+      final serverpod = Serverpod(
+        ['--role', 'maintenance'],
+        Protocol(),
+        Endpoints(),
+        config: config,
+      );
+
+      expect(serverpod.config.role, equals(ServerpodRole.maintenance));
+      // Other values should remain from config
+      expect(serverpod.config.runMode, equals('development'));
+      expect(serverpod.config.serverId, equals('test-server'));
+      expect(serverpod.config.loggingMode, equals(ServerpodLoggingMode.normal));
+      expect(serverpod.config.applyMigrations, isFalse);
+      expect(serverpod.config.applyRepairMigration, isFalse);
+    });
+
+    test(
+        'when applyMigrations is set to false in config and true in CLI args '
+        'then server applies migrations', () {
+      final config = ServerpodConfig(
+        runMode: 'development',
+        serverId: 'test-server',
+        role: ServerpodRole.monolith,
+        loggingMode: ServerpodLoggingMode.normal,
+        applyMigrations: false,
+        applyRepairMigration: false,
+        apiServer: ServerConfig(
+          port: 8080,
+          publicHost: 'localhost',
+          publicPort: 8080,
+          publicScheme: 'http',
+        ),
+        maxRequestSize: 524288,
+      );
+
+      final serverpod = Serverpod(
+        ['--apply-migrations'],
+        Protocol(),
+        Endpoints(),
+        config: config,
+      );
+
+      expect(serverpod.config.applyMigrations, isTrue);
+      // Other values should remain from config
+      expect(serverpod.config.runMode, equals('development'));
+      expect(serverpod.config.serverId, equals('test-server'));
+      expect(serverpod.config.role, equals(ServerpodRole.monolith));
+      expect(serverpod.config.loggingMode, equals(ServerpodLoggingMode.normal));
+      expect(serverpod.config.applyRepairMigration, isFalse);
+    });
+
+    test(
+        'when applyRepairMigration is set to false in config and true in CLI args '
+        'then server applies repair migration', () {
+      final config = ServerpodConfig(
+        runMode: 'development',
+        serverId: 'test-server',
+        role: ServerpodRole.monolith,
+        loggingMode: ServerpodLoggingMode.normal,
+        applyMigrations: false,
+        applyRepairMigration: false,
+        apiServer: ServerConfig(
+          port: 8080,
+          publicHost: 'localhost',
+          publicPort: 8080,
+          publicScheme: 'http',
+        ),
+        maxRequestSize: 524288,
+      );
```

Comment:
Suggestion: Setup a test builder (this might already exist!) and use that instead of constructing the object manually like this. 

The benefit is that if we ever modify the ServerpodConfig object with new fields (especially mandatory fields), these tests will not have to be modified. 
Another benefit is that we can ignore all fields that we are not explicitly testing for, just use the `with` method for the field we want to validate. This shrinks the number of lines in the setup here making it easier to view the full test.